src/include/daos_srv/daos_engine.h

```diff
@@ -803,4 +805,32 @@ dss_chore_diy(struct dss_chore *chore);
 bool
 engine_in_check(void);
 
+/**
+ * @brief Selects the appropriate module version based on the protocol version
+ * for server side rpc.
+ *
+ * @param module_ver    Output parameter for the selected module version
+ * @return 0 or negative error
+ */
+static inline int
+dss_select_module_version(int module_id, uint8_t *module_ver)
+{
+	struct dss_module_info *dmi          = dss_get_module_info();
+	uint8_t                 protocol_ver = daos_version_get_protocol(&dmi->dmi_version);
+
+	return daos_get_module_version(module_id, protocol_ver, module_ver);
+}
+
+#define DEFINE_DS_RPC_PROTOCOL(module_prefix, module_id)                                           \
+	int ds_##module_prefix##_rpc_protocol(uint8_t *version)                                    \
+	{                                                                                          \
+		return dss_select_module_version(module_id, version);                              \
+	}
+
+#define DEFINE_RPC_PROTOCOL(module_prefix, module_id)                                              \
+	int module_prefix##_rpc_protocol(uint8_t *version)                                         \
+	{                                                                                          \
+		return dss_select_module_version(module_id, version);                              \
+	}
+
```

Comment:
What the difference of these two macros?

Reply:
It implement same function, i just implement it to follow different module's name style. Eg some module like pool/container, it prefer ds_pool/cont, some modules only used in the server side, function name avoid ds_prefix there..